sphinx/directives/code.py

```diff
@@ -159,7 +160,7 @@ def run(self):
             try:
                 literal = container_wrapper(self, literal, caption)
             except ValueError as exc:
-                return [document.reporter.warning(str(exc), line=self.lineno)]
+                return [document.reporter.warning(text_type(exc), line=self.lineno)]
```

Comment:
Finally I used `text_type()` to stringify. `"%r"` converts exceptions to `ValueError('blah blah...')`. It's not good for message.